### PR TITLE
add support for remark-lint

### PR DESCRIPTION
**Pull request by rhysd**
I added support for [remark-lint](https://github.com/wooorm/remark-lint) to ale. It is a linter for markdown. I marked this linter as existing file only because stdin or temporary file is not available. remark-lint assumes `.remarkrc` config file put in the same or any parent directory as target markdown source file.

### CODE REVIEW

File: ale_linters/markdown/remark_lint.vim [hunk 1]
```diff
@@ -0,0 +1,30 @@
+" Author rhysd https://rhysd.github.io/
+" Description: remark-lint for Markdown files
+
+function! ale_linters#markdown#remark_lint#Handle(buffer, lines) abort
+    " matches: '  1:4  warning  Incorrect list-item indent: add 1 space  list-item-indent  remark-lint'
+    let l:pattern = '^ \+\(\d\+\):\(\d\+\)  \(warning\|error\)  \(.\+\)$'
+    let l:output = []
+
+    call writefile(a:lines, $HOME . '/this_is_test.txt')
```
Comment (w0rp) [inline ale_linters/markdown/remark_lint.vim]: You left a test line here.

My recommended way to debug jobs and channels is to use `ch_logfile` in Vim 8.

Comment (rhysd) [reply ale_linters/markdown/remark_lint.vim]: ah, sorry...

**Comment by rhysd** [comment]
@w0rp I fixed the point :+1:

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**